lib/FusionInventory/Agent/Task/Inventory/Win32/Softwares.pm

```diff
@@ -411,4 +430,56 @@ sub _processMSIE {
 
 }
 
+# List of SQL Instances
+sub _getSqlInstancesList {
+    my (%params) = @_;
+    my $logger = $params{logger};
+	my $softwareName = $params{softwarename};
+	my $softwareVersion = $params{softwareversion};
+	
+	# Registry access for SQL Instances
+    my $machKey = $Registry->Open('LMachine', {Access => KEY_READ}) or $logger->error("Can't open HKEY_LOCAL_MACHINE key: $EXTENDED_OS_ERROR");
+    my $sqlinstancesList = $machKey->{"SOFTWARE/Microsoft/Microsoft SQL Server/Instance Names/SQL"};
+
```

Comment:
Here, you should use a `return unless $sqlinstancesList;` or you can use the syntax:
```
    my $sqlinstancesList = $machKey->{"SOFTWARE/Microsoft/Microsoft SQL Server/Instance Names/SQL"}
        or return;
```
Otherwise if $sqlinstancesList is undefined at line 445, this will crash the Software module.